code/iaas/allocator/src/main/java/io/cattle/platform/allocator/service/AllocatorServiceImpl.java

```diff
@@ -182,6 +197,39 @@ public void mergeLabels(Map<String, String> srcMap, Map<String, String> destMap)
         return constraints;
     }
 
+    /**
+     * Supported macros
+     * ${service_name}
+     * ${project_name}
+     *
+     * @param valueStr
+     * @param instance
+     * @return
+     */
+    private String evaluateMacros(String valueStr, Instance instance) {
+        if (valueStr.indexOf(SERVICE_NAME_MACRO) != -1 || valueStr.indexOf(PROJECT_NAME_MACRO) != -1) {
+            Service service = null;
+
+            List<? extends Service> services = serviceDao.findServicesFor(instance);
+            if (services.size() > 0) {
+                service = services.get(0);
+            }
+
```

Comment:
nit: we can just check if the service is null, and return immediately if it is. It would eliminate the need for the not null check on line 222.